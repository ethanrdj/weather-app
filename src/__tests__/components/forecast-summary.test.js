import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/forecast-summary";

describe("ForecastSummary", () => {
  xit("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={1525046400000}
        temperature="mockTemperature"
        description="mockDescription"
        icon="800"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  xit("renders the correct props", () => {
    const { getByText } = render(
      <ForecastSummary
        date={1525046400000}
        temperature="mockTemperature"
        description="mockDescription"
        icon="800"
      />
    );

    expect(getByText("Mon 30th Apr")).toHaveClass("date");
    expect(getByText("mockTemperatureÂ°c")).toHaveClass("temperature");
    expect(getByText("mockDescription")).toHaveClass("description");
    expect(getByText("???")).toHaveClass("wi wi-day-sunny icon???");
  });
});
