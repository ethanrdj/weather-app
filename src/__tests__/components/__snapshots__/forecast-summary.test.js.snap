// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="date"
      data-testid="date-id"
    >
      Mon 30th Apr
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      mockTemperatureÂ°c
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      mockDescription
    </div>
    <i
      class="wi wi-day-sunny icon"
      data-testid="icon-id"
      name="owm"
    />
    <button
      value="1525046400000"
    >
      More Details
    </button>
  </div>
</DocumentFragment>
`;